modules/administration-guide/pages/configuring-fuse.adoc

```diff
@@ -0,0 +1,131 @@
+:_content-type: ASSEMBLY
+:description: Configuring fuse-overlayfs
+:keywords: administration-guide, configuring, fuse
+:navtitle: Configuring fuse-overlayfs
+:page-aliases:
+
+[id="configuring-fuse-overlayfs"]
+= Configuring fuse-overlayfs
+
+By default, the Universal Developer Image (UDI) contains Podman which can be used to build and push container images within a workspace.
+However, Podman in the UDI is configured to use the `vfs` storage driver which does not provide copy-on-write support.
+For more efficient image management, it is recommended to use the fuse-overlayfs storage driver which supports copy-on-write in rootless environments.
+
+This page describes how to make `/dev/fuse` be accessible to workspace containers, which is a requirement for using fuse-overlayfs.
+
+== Enabling container access to /dev/fuse for OpenShift
+
+[WARNING]
+====
+Creating `MachineConfig` resources on an OpenShift cluster is a potentially dangerous task, as it allows making advanced, system-level changes to the cluster.
+
+View the link:https://docs.openshift.com/container-platform/{ocp4-ver}/post_installation_configuration/machine-configuration-tasks.html#machine-config-overview-post-install-machine-configuration-tasks[MachineConfig documentation] for more details and possible risks.
+
+====
+
+.Prerequisites
+
+* The link:https://docs.openshift.com/container-platform/{ocp4-ver}/installing/install_config/installing-customizing.html#installation-special-config-butane-install_installing-customizing[Butane] tool (`butane`) is installed in the operating system you are using.
+
+* An active `{orch-cli}` session with administrative permissions to the destination OpenShift cluster. See {orch-cli-link}.
+
+.Procedure
+
+. Set an environment variable depending on whether your OpenShift cluster is a single node cluster, or a multi node cluster with separate control plane and worker nodes.
++
+For a single node cluster, set:
+[subs="+quotes,+attributes,+macros"]
+----
+$ NODE_ROLE=master
+----
+For a multi node cluster, set:
+[subs="+quotes,+attributes,+macros"]
+----
+$ NODE_ROLE=worker
+----
+
+. Create a `MachineConfig` resource that creates a drop-in CRI-O configuration file named `99-podman-fuse` in the `NODE_ROLE` nodes. This configuration file allows access to the `/dev/fuse` device for certain pods.
++
+[subs="+quotes,+attributes,+macros"]
+----
+cat << EOF | butane | oc apply -f -
+variant: openshift
+version: 4.12.0
```

Comment:
Is it OpenShift version?

Reply:
This is the butane

Reply:
This is the OpenShift butane config version: https://coreos.github.io/butane/upgrading-openshift/

Since version 4.12.0 is compatible with 4.13.0 and 4.14.0, I kept the version to 4.12.0

Reply:
For OpenShift >= 4.15, we do not need the MachineConfig to enable fuse overlay. I plan to provide a new PR to update the docs and merge it when OpenShift 4.15 is available